lib/controllers/root-controller.js

```diff
@@ -574,6 +587,44 @@ export default class RootController extends React.Component {
     viewElement.focus();
   }
 
+  viewChangesForCurrentFile(stagingStatus) {
+    const editor = this.props.workspace.getActiveTextEditor();
+    const absFilePath = editor.getPath();
+    const repoPath = this.props.repository.getWorkingDirectoryPath();
+    if (absFilePath.startsWith(repoPath)) {
+      const filePath = absFilePath.slice(repoPath.length + 1);
+      this.quietlySelectItem(filePath, stagingStatus);
+      const splitDirection = this.props.config.get('github.viewChangesForCurrentFileDiffPaneSplitDirection');
+      const pane = atom.workspace.getActivePane();
+      if (splitDirection === 'right') {
+        pane.splitRight();
+      } else if (splitDirection === 'down') {
+        pane.splitDown();
+      }
+      return this.diveIntoFilePatchForPath(filePath, stagingStatus, {
+        lineNumber: editor.getCursorBufferPosition().row + 1,
+        amending: stagingStatus === 'staged' && this.state.isAmending,
+      });
+    } else {
+      throw new Error(`${absFilePath} does not belong to repo ${repoPath}`);
+    }
+  }
+
+  @autobind
+  viewUnstagedChangesForCurrentFile() {
+    this.viewChangesForCurrentFile('unstaged');
+  }
+
+  @autobind
+  viewStagedChangesForCurrentFile() {
+    this.viewChangesForCurrentFile('staged');
+  }
+
+  @autobind
+  goToDiffLine(lineNumber) {
+    return this.filePatchController.goToDiffLine(lineNumber);
+  }
```

Comment:
The way we manage model data doesn't suit itself well to Redux in my opinion. I also wanted to live with explicit props for a while to ensure we didn't switch to something a little more magical unless we really felt the pain and needed to — the abstraction, like any other, has a cost, and already the more meta-programmy parts of the package are the hardest to understand.

I *do* think we could use the other half of Redux — actions on the context — to good effect in the package. For example, anything that lives on the `AtomEnvironment` global could be placed on the context and components that need them could be wrapped in an HoC that accesses them (but allows them to be passed explicitly as props in tests). And other methods that perform actions that do not directly modify state and that do not affect rendering could be passed on context, as long as we're careful and selective about it.